CAIPs/caip-350.md

```diff
@@ -0,0 +1,120 @@
+discussions-to: https://ethereum-magicians.org/t/erc-7930-interoperable-addresses/23365
+status: Draft
+type: Standard
+created: 2025-04-18
+requires: 2
+---
+
+## Simple Summary
+A CAIP profile for every chain namespace to describe how to serialize their addresses and chain references into the Interoperable Address format described in [ERC-7930].
+
+## Abstract
+Every chain defines their own address types & also a way to identify networks within that chain's ecosystem.
+[ERC-7930] is an address format for (address, chain) pairs supporting any chain, and also including metadata on the chain & addresses' type so no information needs to be communicated out of band.
+This CAIP profile aims to be a living & mutable repository of serialization methods for all chains to uniformly conform to Interoperable Addresses.
+
+## Motivation
+Standards like CAIP-2 and CAIP-10 are simple text representations of addresses & chain namespaces/references, but they do not address:
+- Binary representation of addresses and/or chain namespace+reference: Relevant & desired for on-chain usage.
+- Canonicity: CAIP-10, being a generic text format, leaves the canonization and serialization of ASCII address to on-chain addresses to each namespace's profile; unless a given namespace's profile specifies such logic, the standard itself makes no universal guarantees on a blockchain account (_target address_ in ERC-7930 parlance) having only one CAIP-10 representation, which makes them less useful as dictionary keys.
+- Completeness: both formats have limits on the length of their identifiers, which are reasonable for human-readability but often identifiers have to drop meaningful information to conform to those length requirements. While that information might be easy to look up in the context of wallet software, doing so within a smart contract would not be possible.
+- Succinctness: Text formats necessarily have to incur encoding overhead compared to binary ones, causing relative informational inefficiency.
+
+## Specification
+The purpose of every supporting profile is be to be able to deterministically and unambiguously convert between:
+- Customary address text formats of the ecosystem (which may also be described in its [CAIP-10] profile) and the Interoperable Address text representation defined in [ERC-7930]
+- Customary chain reference text formats of the ecosystem (which may also be described in [CAIP-2]) and the text representation defined in [ERC-7930]
+- [ERC-7930]'s Interoperable Address -> [ERC-7930]'s Interoperable Name and back of the ecosystem's chain references
+- [ERC-7930]'s Interoperable Address -> [ERC-7930]'s Interoperable Name and back of the ecosystem's addresses *for all address formats of said ecosystem*
+
+And must also define the binary id of the namespace itself, akin to [ERC-7930]'s `ChainType`
+
+For this, every namespace MUST define the following sections in their definition of their CAIP-350 profile:
```

Comment:
```suggestion
For this, every namespace MUST define specify all of the above in a CAIP-350 profile to maximize interoperability and review. A [template for these profiles](https://github.com/ChainAgnostic/namespaces/blob/main/_template/caip350.md) is defined in the Namespaces registry.
```

And move the template to the URL above by opening a separate PR [over in namespaces](https://github.com/ChainAgnostic/namespaces/pulls)